pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,368 @@
+"""
+Module for Beryllium Lens Calculations
+"""
+from periodictable import xsf
+from periodictable import formula as ptable_formula
+import numpy as np
+import datetime
+import os
+import shutil
+import pprint
+from itertools import product
+
+
+"""
+If you are using the IMS IOC, each lens motor should have a
+base record associated with it  i.e MFX:DIA:XFLS.
+There should also be a binary record for each state,
+that is just this base plus an arbitrary string i.e MFX:DIA:XFLS:OUT
+"""
+
+# We have sets of Be lenses with thicknesses:
+LENS_RADII = [50e-6, 100e-6, 200e-6, 300e-6, 500e-6, 1000e-6, 1500e-6]
+
+
+def getAttLen(E, material="Be", density=None):
+    """ get the attenuation length (in meter) of material (default Si), if no
+      parameter is given for the predefined energy;
+      then T=exp(-thickness/att_len); E in keV"""
+    att_len = float(xsf.attenuation_length(material, density=density,
+                    energy=E))
+    return att_len
+
+
+def get_delta(E, material="Be", density=None):
+    """
+    Calculates delta for a given material at a given energy
+
+    Parameters
+    ----------
+    E: number
+        Beam Energy
+    material : `str`
+        Beryllium. The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density : TODO: find out what density is
+    Returns
+    --------
+    delta: `int` TODO: not sure if int here or float?
+    """
+    delta = 1 - np.real(xsf.index_of_refraction(material, density=density,
+                        energy=E))
+    return delta
+
+
+def calc_focal_length_for_single_lens(E, radius, material="Be", density=None):
+    """
+    Calculates the Focal Length for a single lens.
+
+    Parameters
+    ----------
+    E: number
+        Beam Energy
+    radius : float TODO: is this float or no?
+    material : `str`
+        Beryllium. The use of beryllium extends the range of operation
+        of compound refractive lenses, improving transmission,
+        aperture size, and gain
+    density : TODO: find out what density is
+
+    Returns
+    --------
+    focal_length : `float`
+        The focal length for a single lens
+    """
+    delta = get_delta(E, material, density)
+    focal_length = radius / 2.0 / delta
+    return focal_length
+
+
+def calc_focal_length(E, lens_set, material="Be", density=None):
+    """
+    Calculates the Focal Length for certain lenses configuration and energy.
+
+    Parameters
+    -----------
+    E: number
+        Beam Energy
+    lens_set : `list`
+        [(numer1, lensthick1), (number2, lensthick2)...]
```

Comment:
According to the code it's:
```suggestion
        [numer1, lensthick1, number2, lensthick2...]
```

Reply:
yes that's right, I realized that after some testing :) - thank you @klauer